src/appleseed/renderer/modeling/bssrdf/directionaldipolebssrdf.cpp

```diff
@@ -146,22 +213,131 @@ namespace
             value += tmp;
             value *= 0.5f;
 #endif
+            value *= static_cast<float>(norm(outgoing_point.get_point() - incoming_point.get_point()));
+            value *= static_cast<float>(values->m_weight);
+        }
 
-            // Return r * R(r) * weight.
-            const double radius = norm(incoming_point.get_point() - outgoing_point.get_point());
-            value *= static_cast<float>(radius * values->m_weight);
+        double evaluate_pdf(
+            const void*             data,
+            const size_t            channel,
+            const double            radius) const APPLESEED_OVERRIDE
+        {
+            const DirectionalDipoleBSSRDFInputValues* values =
+                reinterpret_cast<const DirectionalDipoleBSSRDFInputValues*>(data);
+
+            // PDF of the sampled radius.
+            double pdf_radius = 0.0;
+            for (size_t i = 0, e = values->m_sigma_tr.size(); i < e; ++i)
+            {
+                const double sigma_tr = values->m_sigma_tr[i];
+                pdf_radius += exponential_distribution_pdf(radius, sigma_tr);
+            }
+
+            pdf_radius /= static_cast<double>(values->m_sigma_tr.size());
+
+            // PDF of the sampled angle.
+            const double pdf_angle = RcpTwoPi;
+
+            // Compute and return the final PDF.
+            return pdf_radius * pdf_angle;
         }
 
       private:
+        void apply_preset(const size_t preset, DirectionalDipoleBSSRDFInputValues* values) const
+        {
+            switch(preset)
+            {
+                case 1: // Apple
```

Comment:
Indent `case` statements with two spaces.
Indent code below `case` statements with two more spaces.
Indent `break` with the rest of the code.

Example:

    switch (preset)
    {
      case 1:
        ...
        break;